runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -1118,6 +1154,9 @@ private void pruneRemovedRouters(@Nonnull Layout layout) {
                         // Prune away removed node routers from the nodeRouterPool.
                         pruneRemovedRouters(l);
 
+                        getRegistry().ifPresent(reg ->
+                                fetchSample.ifPresent(sample ->
+                                        sample.stop(reg.timer("runtime.fetch_layout.timer"))));
```

Comment:
What is the advantage of using Java's Optional over the No-op metric types that micrometer provides?

Reply:
I did not know about no-op times. But I guess this also depends on the order the registries and the meters are instantiated. As far as I understand, the meters are no-oped unless they are registered. Since we configure the registry on startup first, depending on the passed arguments the registry is optionally getting created first, hence by composition all of the meters that are produced from it are also optional. 
Also having an optional registry allows for a more fluent code when it comes to lookup by key.